data_mgr.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Create and update the data structure for all workflow services."""
+
+from fnmatch import fnmatchcase
+
+from tornado import gen
+# from google.protobuf.json_format import MessageToDict
+
+# from cylc import flags
+from cylc.exceptions import ClientError, ClientTimeout
+from cylc.ws_messages_pb2 import PbEntireWorkflow
+from cylc.network.scan import MSG_TIMEOUT
```

Comment:
We will have to update to `cylc.flow` later.

Reply:
Is it solved in your local branch @dwsutherland ? Just synced branches and noticed I had to alter the imports locally again to get it working.